crates/control/src/ball_filter.rs

```diff
@@ -164,19 +189,26 @@ impl BallFilter {
                 && duration_since_last_observation < filter_parameters.hypothesis_timeout
         };
 
+        // TODO: this removes hypotheses if not one is resting and the other one is moving!
         let should_merge_hypotheses =
-            |hypothesis1: &BallHypothesis, hypothesis2: &BallHypothesis| {
-                let ball1 =
-                    hypothesis1.choose_ball(filter_parameters.resting_ball_velocity_threshold);
-                let ball2 =
-                    hypothesis2.choose_ball(filter_parameters.resting_ball_velocity_threshold);
-
-                distance(ball1.position, ball2.position)
-                    < filter_parameters.hypothesis_merge_distance
+            |_hypothesis1: &BallHypothesis, _hypothesis2: &BallHypothesis| {
+                // distance(
```

Comment:
The problem of merging moving and resting hypotheses could be addressed by simply matching the `BallMode`s of the Hypotheses. In the case of a two resting hypotheses, the positions can simply be compared. In the case of moving hypotheses one could compare the position of the ball hypothesis and the direction as well as magnitude of the velocity.

Reply:
decided to do this only for resting since all moving hypotheses eventually become resting